### PR TITLE
Added initial capacity for collecions, removed some excessive code

### PR DESCRIPTION
**Pull request by DamirAinullin**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio
- [ ] Require translation update
- [ ] Require document update (readme.md, wikipage, etc)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New feature

---

### Description of your *pull request* and other information

Added initial capacity for collecions inpaces where it is possible, removed some excessive code.

### CODE REVIEW

File: shadowsocks-csharp/Controller/Service/AvailabilityStatistics.cs [hunk 1]
```diff
@@ -197,10 +197,8 @@ private void UpdateRecords()
                     _latencyRecords.TryGetValue(id, out latencyRecords);
                     StatisticsRecord record = new StatisticsRecord(id, inboundSpeedRecords, outboundSpeedRecords, latencyRecords);
                     /* duplicate server identifier */
```
Comment (studentmain) [inline shadowsocks-csharp/Controller/Service/AvailabilityStatistics.cs:199]: remove this comment as we no longer check duplicate here

File: shadowsocks-csharp/Controller/Service/AvailabilityStatistics.cs [hunk 2]
```diff
@@ -360,8 +358,7 @@ public void UpdateLatency(Server server, int latency)
         {
             _latencyRecords.GetOrAdd(server.Identifier(), (k) =>
```
Comment (studentmain) [inline shadowsocks-csharp/Controller/Service/AvailabilityStatistics.cs:359]: single line lambda should works

File: shadowsocks-csharp/Controller/ShadowsocksController.cs [hunk 3]
```diff
@@ -557,7 +557,7 @@ protected void Reload()
 
                 TCPRelay tcpRelay = new TCPRelay(this, _config);
                 UDPRelay udpRelay = new UDPRelay(this);
-                List<Listener.IService> services = new List<Listener.IService>
```
Comment (studentmain) [inline shadowsocks-csharp/Controller/ShadowsocksController.cs:560]: If it's for performance, I think we only need add them in critical path.

File: shadowsocks-csharp/Encryption/AEAD/AEADMbedTLSEncryptor.cs [hunk 4]
```diff
@@ -80,19 +80,19 @@ public override void cipherEncrypt(byte[] plaintext, uint plen, byte[] ciphertex
                 case CIPHER_AES:
                     ret = MbedTLS.cipher_auth_encrypt(_encryptCtx,
                         /* nonce */
-                        _encNonce, (uint) nonceLen,
```
Comment (studentmain) [inline shadowsocks-csharp/Encryption/AEAD/AEADMbedTLSEncryptor.cs:83]: we don't have an official code format yet, IMO we should keep their format to reduce diff.

File: shadowsocks-csharp/Util/ViewUtils.cs [hunk 5]
```diff
@@ -25,7 +25,7 @@ public static class ViewUtils
         public static IEnumerable<MenuItem> GetMenuItems(Menu m)
         {
             if (m?.MenuItems == null || m.MenuItems.Count == 0) return Enumerable.Empty<MenuItem>();
-            var children = new List<MenuItem>();
+            var children = new List<MenuItem>(m.MenuItems.Count);
```
Comment (studentmain) [inline shadowsocks-csharp/Util/ViewUtils.cs:28]: maybe some LINQ will solve this easier?

File: shadowsocks-csharp/Controller/ShadowsocksController.cs (hunk 3 above)
Comment (DamirAinullin) [reply shadowsocks-csharp/Controller/ShadowsocksController.cs:560]: I think it is a good practice to set rigth initial capacity wherever it is possible. Even if it is not a critical path the code with rigth initial capacity will work faster and memory traffic will be smaller.

File: shadowsocks-csharp/Util/ViewUtils.cs (hunk 5 above)
Comment (DamirAinullin) [reply shadowsocks-csharp/Util/ViewUtils.cs:28]: I tried bit it does not work for MenuItems

File: shadowsocks-csharp/Encryption/AEAD/AEADMbedTLSEncryptor.cs (hunk 4 above)
Comment (DamirAinullin) [reply shadowsocks-csharp/Encryption/AEAD/AEADMbedTLSEncryptor.cs]: reverted code format as it was before

File: shadowsocks-csharp/Controller/Service/AvailabilityStatistics.cs (hunk 1 above)
Comment (DamirAinullin) [reply shadowsocks-csharp/Controller/Service/AvailabilityStatistics.cs]: done

**Comment by DamirAinullin** [comment]
@studentmain Is it OK now?

**Comment by database64128** [comment]
Stale with mostly unnecessary changes. Closing.

**Pull request closed without merging**